server/src/domain/album/album.service.ts

```diff
@@ -69,18 +70,28 @@ export class AlbumService {
     // Get asset count for each album. Then map the result to an object:
     // { [albumId]: assetCount }
     const albumsAssetCount = await this.albumRepository.getAssetCountForIds(albums.map((album) => album.id));
-    const albumsAssetCountObj = albumsAssetCount.reduce((obj: Record<string, number>, { albumId, assetCount }) => {
-      obj[albumId] = assetCount;
-      return obj;
-    }, {});
+    const albumsAssetCountObj: Record<string, AlbumAssetCount> = albumsAssetCount.reduce(
```

Comment:
If we're going to update this stuff, we should rename it to `getMetadataForIds` instead, and call it "album metadata" everywhere.